python/link_prediction.py

```diff
@@ -0,0 +1,771 @@
+        features_size_loaded = True
+
+
+def _process_help_function(
+    ind: int,
+    old_index: int,
+    old_to_new: Dict[int, int],
+    new_to_old: Dict[int, int],
+    features: List,
+    node_features: List[int],
+):
+    """Helper function for _get_dgl_graph_data
+
+    Args:
+        ind (int): Index vertex counter.
+        old_index (int): Original index
+        old_to_new (Dict[int, int]): mappings from old to new indexes.
+        new_to_old (Dict[int, int]): mappings from new to old indexes.
+        features (List): Saves all features.
+        node_features (List[int]): Node features.
+
+    Returns:
+       None
+    """
+    if old_index not in old_to_new.keys():
+        new_to_old[ind] = old_index
+        old_to_new[old_index] = ind
+        features.append(node_features)
+        return ind + 1
+    return ind
+
+
+def _get_dgl_graph_data(
+    ctx: mgp.ProcCtx,
+) -> Tuple[dgl.graph, Dict[int32, int32], Dict[int32, int32]]:
+    """Creates dgl representation of the graph.
+
+    Args:
+        ctx (mgp.ProcCtx): The reference to the context execution.
+
+    Returns:
+        Tuple[dgl.graph, Dict[int32, int32], Dict[int32, int32]]: Tuple of DGL graph representation, dictionary of mapping new to old index and dictionary of mapping old to new index.
+    """
+
+    global link_prediction_parameters
+
+    src_nodes, dest_nodes = [], []  # for saving the edges
+
+    new_to_old = dict()  # map of new node index to old node index
+    old_to_new = dict()  # map of old node index to new node index
+    features = []  # map of new node index to its feature
+    ind = 0
+
+    for vertex in ctx.graph.vertices:
+        # Process source vertex
+        src_id = vertex.id
+        src_features = vertex.properties.get(link_prediction_parameters.node_features_property)
+        ind = _process_help_function(ind, src_id, old_to_new, new_to_old, features, src_features)
+
+        for edge in vertex.out_edges:
+            # Process destination vertex next
+            dest_node = edge.to_vertex
+            dest_id = dest_node.id
+            dest_features = dest_node.properties.get(link_prediction_parameters.node_features_property)
+            ind = _process_help_function(ind, dest_id, old_to_new, new_to_old, features, dest_features)
+
+            # Create dgl graph
+            src_nodes.append(old_to_new[src_id])
+            dest_nodes.append(old_to_new[dest_id])
+
+    # print("Src nodes: ", src_nodes)
+    # print("Dest nodes: ", dest_nodes)
+    # print("Features: ", features)
+    # print("Ind: ", ind)
+    features = torch.tensor(features, dtype=torch.float32)  # use float for storing tensor of features
+    g = dgl.graph((src_nodes, dest_nodes), num_nodes=ind)
+    g.ndata[link_prediction_parameters.node_features_property] = features
+    g = dgl.add_self_loop(g) 
+    return g, new_to_old, old_to_new
+
+
+def _validate_user_parameters(parameters: mgp.Map) -> Tuple[bool, str]:
```

Comment:
Why this function returns a status code, while others simply throw an exception. From what I saw the exeption is thrown if the status code specifies that something is wrong?